envoy/http/codec.h

```diff
@@ -363,6 +363,11 @@ class Stream : public StreamResetHandler {
    * Get the bytes meter for this stream.
    */
   virtual const StreamInfo::BytesMeterSharedPtr& bytesMeter() PURE;
+
+  /**
+   * Fired when the Envoy level stream representation has been destroyed.
```

Comment:
If we keep this, can we converge on a consistent name/comment? Is this when the stream is complete? Or destroyed for any reason such as reset? So I would either fix the comment or make it `onEnvoyStreamDestroyed()`